Nil pointer when getting context's flagSet in UsageError function.

## my urfave/cli version is

_**v2.3.0**_

## Checklist

- [x] Are you running the latest v2 release? The list of releases is [here](https://github.com/urfave/cli/releases).
- [x] Did you check the manual for your release? The v2 manual is [here](https://github.com/urfave/cli/blob/master/docs/v2/manual.md)
- [x] Did you perform a search about this problem? Here's the [Github guide](https://help.github.com/en/github/managing-your-work-on-github/using-search-to-filter-issues-and-pull-requests) about searching.

## Dependency Management

- [x] My project is using go modules.
- [ ] My project is using vendoring.
- [x] My project is automatically downloading the latest version.
- [ ] I am unsure of what my dependency management setup is.

## Describe the bug

When I try to get flag from UsageError function, program panic because cli.Context's flagSet is a nil pointer.

## To reproduce

```go
package main

import (
	"fmt"

	"github.com/urfave/cli/v2"
)

func main() {
	app := cli.NewApp()
	app.Commands = []*cli.Command{
		{
			Name: "test",
			Flags: []cli.Flag{
				&cli.StringFlag{
					Name: "flag",
				},
			},
			OnUsageError: UsageError,
		},
	}
	err := app.Run([]string{"@app", "test", "--flag", "1", "--notflag", "1"})
	fmt.Println(err)
}

func UsageError(ctx *cli.Context, err error, isSubcommand bool) error {
	flag := ctx.String("flag")
	fmt.Println(flag)
	return nil
}
```

## Observed behavior

![image](https://user-images.githubusercontent.com/45025194/148929909-ec1ff209-c93d-4712-9de2-dcd8464d00d2.png)

panic: runtime error: invalid memory address or nil pointer dereference

[signal SIGSEGV: segmentation violation code=0x1 addr=0x28 pc=0x114b8aa]

## Expected behavior

I expect cli.Context can Lineage correctly and find flag I need.

## Additional context

## Want to fix this yourself?

We'd love to have more contributors on this project! If the fix for this bug is easily explained and very small, free free to create a pull request for it.

## Run `go version` and paste its output here

```
go version go1.16 darwin/amd64
```

## Run `go env` and paste its output here

```
# paste `go env` output in here
```

--- a/context.go
+++ b/context.go
@@ -151,6 +151,9 @@ func (ctx *Context) lookupFlag(name string) Flag {
 
 func (ctx *Context) lookupFlagSet(name string) *flag.FlagSet {
 	for _, c := range ctx.Lineage() {
+		if c.flagSet == nil {
+			continue
+		}
 		if f := c.flagSet.Lookup(name); f != nil {
 			return c.flagSet
 		}

--- a/context_test.go
+++ b/context_test.go
@@ -112,6 +112,8 @@ func TestContext_String(t *testing.T) {
 	c := NewContext(nil, set, parentCtx)
 	expect(t, c.String("myflag"), "hello world")
 	expect(t, c.String("top-flag"), "hai veld")
+	c = NewContext(nil, nil, parentCtx)
+	expect(t, c.String("top-flag"), "hai veld")
 }
 
 func TestContext_Path(t *testing.T) {

